packages/application-components/src/components/tab-header/tab-header.tsx

```diff
@@ -0,0 +1,114 @@
+import { Link, matchPath, useLocation } from 'react-router-dom';
+import startCase from 'lodash/startCase';
+import { useIntl, type MessageDescriptor } from 'react-intl';
+import type { ReactNode } from 'react';
+import type { LocationDescriptor } from 'history';
+import { warning } from '@commercetools-uikit/utils';
+import Text from '@commercetools-uikit/text';
+import { getTabHeaderStyles, getLinkStyles } from './tab.styles';
+
+const pathWithoutSearch = (path: TTabHeaderProps['to']) =>
+  typeof path === 'string' ? path.split('?')[0] : path.pathname;
+
+const warnIfMissingContent = (props: TTabHeaderProps) => {
+  const hasContent = Boolean(props.intlMessage) || Boolean(props.label);
+
+  warning(
+    hasContent,
+    'TabHeader: one of either `label` or `intlMessage` is required but their values are `undefined`'
+  );
+};
+
+const getDisabledTabHeaderAriaAttributes = (
+  isDisabled: TTabHeaderProps['isDisabled']
+) => (isDisabled ? { 'aria-disabled': true } : {});
+
+const getDisabledLinkAtributes = (isDisabled: TTabHeaderProps['isDisabled']) =>
+  isDisabled ? { tabIndex: -1 } : {};
+
+export type TTabHeaderProps = {
+  /**
+   * A route path to redirect to when the tab is clicked.
+   */
+  to: string | LocationDescriptor;
+  /**
+   * The label of the tab.
+   * * <br />
+   * Required if `intlMessage` is not provided.
+   */
+  label?: string;
+  /**
+   * The label of the tab, using an `intl` message object.
+   * <br />
+   * Required if `label` is not provided.
+   */
+  intlMessage?: MessageDescriptor & {
+    values?: Record<string, ReactNode>;
+  };
+  /**
+   * If `true`, indicates that the element is in a disabled state.
+   */
+  isDisabled: boolean;
+  /**
+   * If `true`, marks the tab as active if the link matches exactly the route.
+   */
+  exactPathMatch: boolean;
+};
+
+export const TabHeader = (props: TTabHeaderProps) => {
+  const intl = useIntl();
+  const location = useLocation();
+  const isActive = Boolean(
+    matchPath(location.pathname, {
+      // strip the search, otherwise the path won't match
+      path: pathWithoutSearch(props.to),
+      exact: props.exactPathMatch,
+      strict: false,
+    })
+  );
+  const isDisabled = props.isDisabled;
+
+  let label = props.label;
+  if (props.intlMessage) {
+    label = intl.formatMessage(props.intlMessage);
+  }
+
+  const dataAttributeProps = {
+    'data-track-event': 'click',
+    ...(props.label && {
+      'data-track-component': startCase(label),
+    }),
+  };
```

Comment:
Just wanted to double check if is it ok to only add the `data-track-component` attribute when the consumer used the `label` property.
Perhaps we could also check, in case the consumer used the `intlMessage` prop, if `id` attribute exists in it.

Reply:
This was meant to be using the `label` variable. However, the `label` itself can be technically `undefined` even though we do log a warning if that's the case.

```suggestion
  const dataAttributeProps = {
    'data-track-event': 'click',
    ...(label && {
      'data-track-component': startCase(label),
    }),
  };
```